### PR TITLE
docs on cpu vs. io bound blocking

### PR DESCRIPTION
**Pull request by tanner0101**

### CODE REVIEW

**Comment by tanner0101** [comment]
cc @mcdappdev

File: 4.0/docs/async.md [hunk 1]
```diff
@@ -312,9 +307,23 @@ router.get("hello") { req -> Future<String> in
...
+
+While an event loop is processing CPU bound work, it will be unable to respond to other incoming requests. This is normally fine since CPUs are fast and most CPU work web applications do is lightweight. But this can become a problem if routes with long running CPU work are preventing requests to faster routes from being responded to quickly. 
+
+Identifying long running CPU work in your app and moving it to background threads can help improve the reliability and responsiveness of your service. CPU bound work is more of a gray area than I/O bound work, and it is ultimately up to you to determine where you want to draw the line. 
+
+A common example of heavy CPU bound work is Bcrypt hashing during user signup and login. Bcrypt is deliberately very slow and CPU intensive for security reasons. This may be the most CPU intensive work a simple web application actually does. Moving hashing to a background thread can allow the CPU to interleave event loop work while calculating hashes which results in higher concurrency.
```
Comment (madsodgaard) [inline 4.0/docs/async.md]: If this is "recommended" and occurs in so many apps, maybe an example of how you would do it inside a route handler would be nice?

Comment (0xTim) [reply 4.0/docs/async.md]: Ideally Vapor's BCrypt should do this all for you

Comment (tanner0101) [reply 4.0/docs/async.md]: There's an example here: https://github.com/vapor/vapor/blob/master/Tests/VaporTests/AuthenticationTests.swift#L152-L192

There's an issue open to consider how to include this in Vapor. I'm going to be looking into that after release.

**Pull request merged**